File: tests/java/tests/java/org/elastos/hive/BackupSubscriptionTest.java
package org.elastos.hive;

import org.elastos.hive.config.TestData;
import org.elastos.hive.subscription.PricingPlan;
import org.junit.jupiter.api.*;

import java.util.List;

@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
class BackupSubscriptionTest {
	private static BackupSubscription subscription;

	@BeforeAll public static void setup() {
		Assertions.assertDoesNotThrow(()->{
			TestData testData = TestData.getInstance();
			subscription = new BackupSubscription(
					testData.getAppContext(),
					testData.getProviderAddress());
		} );
	}

	@Test @Order(1) void testGetPricingPlanList() {
		Assertions.assertDoesNotThrow(()->{
			List<PricingPlan> plans = subscription.getPricingPlanList().get();
			Assertions.assertNotNull(plans);
			Assertions.assertFalse(plans.isEmpty());
		});
	}

	@Test @Order(2) void testGetPricingPlan() {
		Assertions.assertDoesNotThrow(()-> {
			PricingPlan plan = subscription.getPricingPlan(VaultSubscriptionTest.PRICING_PLAN_NAME).get();
			Assertions.assertNotNull(plan);
			Assertions.assertEquals(plan.getName(), VaultSubscriptionTest.PRICING_PLAN_NAME);
		});
	}

	@Disabled
	@Test @Order(3) void testSubscribe() {
		Assertions.assertDoesNotThrow(()->subscription.subscribe().get());
	}

	@Disabled
	@Test @Order(6) void testUnsubscribe() {
		Assertions.assertDoesNotThrow(()->subscription.unsubscribe().get());
	}
}

File: tests/src/subscribe/backupsubscription.test.ts
import { PricingPlan, VaultInfo, BackupSubscriptionService } from "@dchagastelles/elastos-hive-js-sdk";
import { ClientConfig } from "../config/clientconfig";
import { TestData } from "../config/testdata";

describe("test backup subscription function", () => {

    let testData: TestData;
    let backupSubscriptionService: BackupSubscriptionService;
    let PRICING_PLAN_NAME: string = "Rookie";

    beforeEach(async () => {
        testData = await TestData.getInstance(ClientConfig.CUSTOM, TestData.USER_DIR);
        backupSubscriptionService = new BackupSubscriptionService(
            testData.getAppContext(),
            testData.getProviderAddress());
        });
        
    test("get pricing plans", async() => {
        let plans: PricingPlan[] = await backupSubscriptionService.getPricingPlanList();
        expect(plans).not.toBeNull();
        expect(plans.length).toBeGreaterThan(0);
    });
    
    test("get pricing plan", async() => {
        let plan: PricingPlan = await backupSubscriptionService.getPricingPlan(PRICING_PLAN_NAME);
        expect(plan).not.toBeNull();
        expect(plan.getName()).toBe(PRICING_PLAN_NAME);
    });

    test("should return vault info", async () => {
        let vaultInfo : VaultInfo = await backupSubscriptionService.subscribe();
        expect(vaultInfo).not.toBeNull();
    });

    test.skip("should unsubscribe", async () => {
        expect(await backupSubscriptionService.unsubscribe()).not.toThrow();
    });
        
    test.skip("Check subscription", async() => {
        let info: VaultInfo = await backupSubscriptionService.checkSubscription();
        expect(info).not.toBeNull();
        console.log(JSON.stringify(info));

    });
    
});


